### PR TITLE
[Feature][Bugfix][Spec Decode][v0.26.0rc] Support DSpark with LMHead TP

### PR DESCRIPTION
**Pull request by jiajinzhu2**
### What this PR does / why we need it?

Backport of #15495 to `releases/v0.26.0rc`.

This PR fixes DSpark speculative decoding when fine-grained LMHead tensor parallelism is enabled.

- Keep the DSpark Markov embedding and low-rank vocabulary projection replicated. Markov decoding is serial across draft positions, so sharding these weights through the LMHead TP group adds an unintended collective to every Markov step.
- Pad the DSpark LMHead input with the draft width `K`, instead of the target runner decode-query width `K + 1`.
- Remove flat dummy sampling rows immediately after the LMHead TP collective and before reshaping logits to `[batch, K, vocab]`.

Without these changes, DSpark uses the generic LMHead TP path with incompatible draft-side shapes and dummy request blocks can reach Markov decoding.

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

```
============lmheadtp=4  ==========================	============ No LMheadtp ============
Successful requests:                     100       	Successful requests:                     100       
Failed requests:                         0         	Failed requests:                         0         
Maximum request concurrency:             4         	Maximum request concurrency:             4         
Benchmark duration (s):                  139.06    	Benchmark duration (s):                  157.02    
Total input tokens:                      26148     	Total input tokens:                      26148     
Total generated tokens:                  30999     	Total generated tokens:                  30986     
Request throughput (req/s):              0.72      	Request throughput (req/s):              0.64      
Output token throughput (tok/s):         222.92    	Output token throughput (tok/s):         197.34    
Peak output token throughput (tok/s):    84.00     	Peak output token throughput (tok/s):    85.00     
Peak concurrent requests:                7.00      	Peak concurrent requests:                7.00      
Total token throughput (tok/s):          410.95    	Total token throughput (tok/s):          363.87    
---------------Time to First Token----------------	---------------Time to First Token----------------
Mean TTFT (ms):                          561.34    	Mean TTFT (ms):                          1271.84   
Median TTFT (ms):                        449.57    	Median TTFT (ms):                        439.87    
P50 TTFT (ms):                           449.57    	P50 TTFT (ms):                           439.87    
P75 TTFT (ms):                           501.61    	P75 TTFT (ms):                           476.26    
P90 TTFT (ms):                           753.99    	P90 TTFT (ms):                           783.11    
P99 TTFT (ms):                           2793.63   	P99 TTFT (ms):                           19022.78  
-----Time per Output Token (excl. 1st token)------	-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          15.77     	Mean TPOT (ms):                          16.10     
Median TPOT (ms):                        15.54     	Median TPOT (ms):                        15.37     
P50 TPOT (ms):                           15.54     	P50 TPOT (ms):                           15.37     
P75 TPOT (ms):                           17.20     	P75 TPOT (ms):                           17.05     
P90 TPOT (ms):                           18.74     	P90 TPOT (ms):                           18.68     
P99 TPOT (ms):                           22.27     	P99 TPOT (ms):                           34.44     
---------------Inter-token Latency----------------	---------------Inter-token Latency----------------
Mean ITL (ms):                           65.76     	Mean ITL (ms):                           65.75     
Median ITL (ms):                         55.39     	Median ITL (ms):                         54.07     
P50 ITL (ms):                            55.39     	P50 ITL (ms):                            54.07     
P75 ITL (ms):                            56.83     	P75 ITL (ms):                            55.41     
P90 ITL (ms):                            58.00     	P90 ITL (ms):                            59.54     
P99 ITL (ms):                            388.78    	P99 ITL (ms):                            377.98    
----------------End-to-end Latency----------------	----------------End-to-end Latency----------------
Mean E2EL (ms):                          5502.58   	Mean E2EL (ms):                          6214.29   
Median E2EL (ms):                        5262.48   	Median E2EL (ms):                        5209.53   
P50 E2EL (ms):                           5262.48   	P50 E2EL (ms):                           5209.53   
P75 E2EL (ms):                           7404.58   	P75 E2EL (ms):                           7879.40   
P90 E2EL (ms):                           9141.73   	P90 E2EL (ms):                           9302.35   
P99 E2EL (ms):                           10300.88  	P99 E2EL (ms):                           25240.52  
---------------Speculative Decoding---------------	---------------Speculative Decoding---------------
Acceptance rate (%):                     63.68     	Acceptance rate (%):                     63.59     
Acceptance length:                       4.18      	Acceptance length:                       4.18      
Drafts:                                  7414      	Drafts:                                  7417      
Draft tokens:                            37070     	Draft tokens:                            37085     
Accepted tokens:                         23605     	Accepted tokens:                         23583     
Per-position acceptance (%):	Per-position acceptance (%):
  Position 0:                            88.49     	  Position 0:                            88.36     
  Position 1:                            74.75     	  Position 1:                            74.73     
  Position 2:                            62.64     	  Position 2:                            62.61     
  Position 3:                            51.50     	  Position 3:                            51.50     
  Position 4:                            41.00     	  Position 4:                            40.74     
==================================================	==================================================
```
- vLLM main: https://github.com/vllm-project/vllm/commit/d02df748bf9efd99022f1a062597dc3cb3808485

### CODE REVIEW

**Review by drslark — APPROVED** [review]
LGTM.
